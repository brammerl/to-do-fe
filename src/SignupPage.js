import React, { Component } from 'react'
import request from 'superagent'

export default class SignupPage extends Component {
    state = {
        email: '',
        password:''
    }


    submitHandle = async(e) => {
        e.preventDefault();

        const token = await request.post(`https://peaceful-anchorage-51064.herokuapp.com/auth/signup`, this.state)
        console.log(token);
        this.props.tokenChange(token.body.token);

    }
    render() {
        console.log(this.state);
        return (
            <div>
                <form onSubmit={this.submitHandle}>
                    <label>
                        Email
                        <input onChange={(e) => this.setState({email: e.target.value})} value={this.state.email}/>
                    </label>
                    <label>
                        Password
                        <input onChange={(e) => this.setState({password: e.target.value})}/>
                    </label>
                    <button>SignUp</button>
                </form>
            </div>
        )
    }
}
