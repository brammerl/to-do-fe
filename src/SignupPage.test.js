import React from 'react';
import { shallow } from 'enzyme';
import SignupPage from './SignupPage.js';

test('renders signup correctly', () => {
  const wrapper = shallow(<SignupPage />);
  expect(wrapper).toMatchSnapshot();
});