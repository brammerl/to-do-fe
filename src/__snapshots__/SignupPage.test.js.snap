// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signup correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        onChange={[Function]}
        value=""
      />
    </label>
    <label>
      Password
      <input
        onChange={[Function]}
      />
    </label>
    <button>
      SignUp
    </button>
  </form>
</div>
`;
